python/cudf/cudf/_libxx/column.pyx

```diff
@@ -162,7 +162,13 @@ cdef class Column:
     @property
     def mask(self):
         if self._mask is None:
-            if self.offset == 0 or self.base_mask is None:
+            if self.base_mask is None or (
+                self.offset == 0
+                and self.base_mask.size == calc_chunk_size(
+                    self.size,
+                    mask_bitsize
+                )
+            ):
```

Comment:
When does this come up? Maybe I'me missing something obvious..

Reply:
If the offset is 0 and the mask is already sized properly according to the column size then we don't want to go and create a newly allocated mask.

This is to guard against something like having a column of size 100 with 0 offset and slicing it down to the first 5 elements, where the new offset is 0 and the new size is 5. The mask should now be smaller than previously so we need to potentially calculate a new mask from the base_mask. But say you slice it from size 100 to size 99, the mask size stays the same so we don't need to calculate a new one and can just return a reference to the base_mask.

There's definitely some future optimizations possible here along the lines of:
- Just resizing the mask when possible (size is reduced but offset stayed 0)
- Using pointer arithmetic if offset is a multiple of 8 and the mask is large enough to accommodate the size properly.

Reply:
Connected offline and there's a known issue here that we'll address in a follow up PR: cuDF Python doesn't currently guarantee the mask is byte padded to 64 bytes (or even 4 bytes) and this is known to break the `copy_bitmask` function.